extensions/ql-vscode/src/databases/database-fetcher.ts

```diff
@@ -429,6 +445,11 @@ async function getStorageFolder(storagePath: string, urlStr: string) {
     counter++;
     folderName = join(realpath, `${lastName}-${counter}`);
     if (counter > 100) {
+      // If there are more than 100 similarly named databases,
+      // give up on using a counter and use a random string instead.
+      folderName = join(realpath, `${lastName}-${nanoid()}`);
+    }
+    if (counter > 200) {
       throw new Error("Could not find a unique name for downloaded database.");
```

Comment:
Although this error shouldn't happen anymore if we use `nanoid`, shall we still update this error message to indicate that the problem is to do with having too many databases?

Maybe just:
```suggestion
      throw new Error("Could not find a unique name for downloaded database. Please remove some databases and try again.");
```
?